### ISSUE TITLE
Фиксация участников турнира

### ISSUE THREAD

**Issue by YuriSilenok** [Students-of-the-city-of-Kostroma/tournament-project#170]
При клике на кнопку  `- - >` в окне регистрации участников, необходимо показать модальнле окно с предупреждением "Если вы нажмете кнопку `Да`, то невозможно будет внести изменения в список участников". Необходимо предусмотреть возможность хранить в БД стадии регистрации участников: регистрация открыта, регистрация закрыта. При открытии приложения, в случае если список участников зафиксирован, необходимо пропустить окно Регистрации участников и перейти к окну Формирование сеток.

**Comment by YuriSilenok** [comment]
@stroywer ,пожалуйста, рассмотрите это это описание на предмет возможности, если оно валидное, внесите его в спецификацию.

**Comment by stroywer** [comment]
А разве такое нужно? Просто по идее пока не нажата кнопка "сетка" в окне формирования групп, мы спокойно можем добавлять ещё участников. И формировать с новыми участниками новые подгруппы.

**Comment by YuriSilenok** [comment]
На данный эта кнопка никак не описана в спецификации. Необходимо уточнить это поведение.

**Comment by stroywer** [comment]
Хорошо, сделаю.

**Comment by stroywer** [comment]
А по поводу зафиксированного списка участников, при котором пропускается окно регистрации участников - всё равно может случиться так, что что-то в списке нужно будет поменять, по этому у пользователя должна быть возможность просмотреть даже и сформированный список участников.

**Comment by YuriSilenok** [comment]
> А по поводу зафиксированного списка участников, при котором пропускается окно регистрации участников - всё равно может случиться так, что что-то в списке нужно будет поменять, по этому у пользователя должна быть возможность просмотреть даже и сформированный список участников.

Я бы не хотел что бы мы додумывали за заказчика. Нам нужно обоснованное поведение.

**Comment by stroywer** [comment]
Тогда нужно спросить самого заказчика - понравиться ли ему такое поведение программы при сформированном списке участников.

**Comment by YuriSilenok** [comment]
> Тогда нужно спросить самого заказчика - понравиться ли ему такое поведение программы при сформированном списке участников.

Да, в нашей команде заказчика представляет Игорь.

**Comment by stroywer** [comment]
Игорь - главный аналитик и тот, и он единственный ведёт переговоры с заказчиком. Не так ли ? Или я что-то путаю?

**Comment by igorslepko** [comment]
> то что-то в списке нужно будет поменять, по этому у пользователя должна быть возможность просмотреть даже и сформированный список участников.

1. посмотреть список он сможет , т.к мы в окне сетки отображаем список участников , разбитый на подгруппы + тоже самое можно будет в окне статистики увидеть
2. если нужно прямо чтото переделать - пусть начинает новый турнир, это не супер проблема (+ возможно мы успеем сделать драг-н-дроп)

**Comment by stroywer** [comment]
#### 1.При клике на кнопку - - > в окне регистрации участников, необходимо показать модальнле окно с предупреждением "Если вы нажмете кнопку Да, то невозможно будет внести изменения в список участников". 
-Она уже прописана в спеке второй страницы первого модуля 8 кнопки “Перейти к турнирной сетке”
#### 2.Необходимо предусмотреть возможность хранить в БД стадии регистрации участников: регистрация открыта, регистрация закрыта. 
-Для чего именно это нужно? Если после регистрации участников и нажатия кнопки “Перейти к турнирной сетке” новых участников не будет возможности добавлять.
#### 3.При открытии приложения, в случае если список участников зафиксирован, необходимо пропустить окно Регистрации участников и перейти к окну Формирование сеток.
-Вы это хотите видеть наподобие того, что когда пользователь регистрирует  турнир, у него высвечивалось сообщение “Не хотите ли сразу добавить сформировавшиеся список участников”, и если пользователь выберет кнопку “Да”, то у него просто появится окно выбора файла типа xls c полным списком участников? Он выбирает данный файл и сразу переходит на страницу формирования подгрупп?

**Comment by YuriSilenok** [comment]
> -Для чего именно это нужно? Если после регистрации участников и нажатия кнопки “Перейти к турнирной сетке” новых участников не будет возможности добавлять.

это необходимо на случай когда пользователь закроет и откроет приложение при клике на турнир он должен или продолжить регистрацию или проскочить регистрацию. Для этого нужно знать, завершена регистрация или нет.

**Comment by YuriSilenok** [comment]
> -Вы это хотите видеть наподобие того, что когда пользователь регистрирует турнир, у него высвечивалось сообщение “Не хотите ли сразу добавить сформировавшиеся список участников”, и если пользователь выберет кнопку “Да”, то у него просто появится окно выбора файла типа xls c полным списком участников? Он выбирает данный файл и сразу переходит на страницу формирования подгрупп?

Нет, я хочу что бы когда регистрация завершена, не было возможности перейти на это окно, даже после переоткрытия приложения

**Comment by stroywer** [comment]
По пункту 2 и по пункту 3, я так думаю, на стартовой странице регистрации турнира нужна рядом с кнопкой "Начать турнир" добавить ещё кнопку "Продолжить турнир (название турнира)". А уже от того, на какой стадии было прекращение прошлого сеанса будет зависеть процесс открытия страницы после нажатия данной кнопки.

**Comment by YuriSilenok** [comment]
> добавить ещё кнопку "Продолжить турнир (название турнира)"

Удобнее будет если сделать список турниров кликабельным. Есть список, мы кликнули на его элемент и продолжили турнир.
